src/@next/components/organisms/CartRow/CartRow.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from "react";
+
+import { Icon, IconButton } from "@components/atoms";
+import { CachedImage, TextField } from "@components/molecules";
+
+import * as S from "./styles";
+import { IProps } from "./types";
+
+const QuantityButtons = (
+  add: () => void,
+  substract: () => void,
+  index?: number
+) => (
+  <S.QuantityButtons>
+    <div
+      onClick={substract}
+      data-cy={`cartPageItem${index}QuantityBtnSubtract`}
+    >
+      <Icon size={16} name="horizontal_line" />
+    </div>
+    <div onClick={add} data-cy={`cartPageItem${index}QuantityBtnAdd`}>
+      <Icon size={16} name="plus" />
+    </div>
+  </S.QuantityButtons>
+);
+
+/**
+ * Product row displayed on cart page
+ */
+export const CartRow: React.FC<IProps> = ({
+  index,
+  totalPrice,
+  unitPrice,
+  name,
+  sku,
+  quantity,
+  maxQuantity,
+  onQuantityChange,
+  thumbnail,
+  attributes = [],
+  onRemove,
+}: IProps) => {
+  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [isTooMuch, setIsTooMuch] = useState(false);
+
+  const handleBlurQuantityInput = () => {
```

Comment:
Not sure if got this function right:

```
const parsed = parseInt(tempQuantity, 10)
let newQuantity = parsed === undefined ?  0 : parsed;

newQuantity = newQuantity > maxQuantity ? maxQuantity : newQuantity;
newQuantity = newQuantity < 0 ? 0 : newQuantity;

setTempQuantity(newQuantity);
if (newQuantity !== parsed) onQuantityChange(maxQuantity);

setIsTooMuch(false);
```

Reply:
Logically yes, except that `setTempQuantity` is only called when value is bigger then max or lower/equal then 0, not always, and `onQuantityChange` is only called when quantity exceeds max and is set to max. The problem which I see I made is that `onQuantityChange` is called also in `handleQuantityChange` when proper value is passed, which would also delay call of `onQuantityChange`, i.e. effectively delegate making decision if `onQuantityChange` has to be called completely to `handleBlurQuantityInput` which I guess would solve the problem and simplify validation, as it will exists only in one place.

Reply:
Ok, I have refactored it, hope it's more clear now.